### PR TITLE
Add custom overlay names with default timestamp placeholder name

### LINKED ISSUE #9: Ability to name themes
> Exporting would be even nicer if we could type the name of the theme before creating a file name.
_Originally posted by @bhajneet in https://github.com/ShabadOS/theme-tool/pull/1#issuecomment-623015497_

### PR DESCRIPTION
**Pull request by saihaj**
Closes #9 
* [x] need to fix saving with empty spaces (need help ... this has to do with #23?? I 🤔 think so ...will wait for any new changes there) 

This branch is based from https://github.com/saihaj/theme-tool/tree/fix-12 so need to get #23 approved first (so making this draft)

### CODE REVIEW

File: src/lib/utils.js [hunk 1]
```diff
@@ -73,3 +73,5 @@ export const loadCss = () => {
       document.documentElement.style.setProperty( storageKey, localStorage[storageKey] )
     ) )
 }
+
+export const timestamp = () => new Date( Date.now() ).toISOString().replace( /\.\d{3}\w$/, '' ).replace( 'T', ' ' )
```
Comment (harjot1singh) [inline src/lib/utils.js]: `new Date()` by default uses the current date and time

```suggestion
export const timestamp = () => new Date().toISOString().replace( /\.\d{3}\w$/, '' ).replace( 'T', ' ' )
```

File: src/components/editors/ExportEditor.js [hunk 2]
```diff
@@ -41,6 +38,20 @@ const ExportEditor = () => {
 
   return (
     <Box p={2}>
+
+      <TextField
+        id="standard-basic"
+        label="Theme Name"
+        placeholder={themeName}
+        onChange={( { target: { value } } ) => {
+          // Empty string then set defualt (Overlay 2020-05-21 HH_MM_SS)
```
Comment (harjot1singh) [inline src/components/editors/ExportEditor.js]:
```suggestion
          // Empty string then set default (Overlay 2020-05-21 HH_MM_SS)
```

Incorrect comment

File: src/components/editors/ExportEditor.js [hunk 3]
```diff
@@ -41,6 +38,20 @@ const ExportEditor = () => {
...
+          // Else User given string
+          if ( value.trim() === '' ) {
+            setThemeName( `Overlay ${timestamp()}` )
+          } else setThemeName( value )
```
Comment (harjot1singh) [inline src/components/editors/ExportEditor.js]: You should either brace both, or one-line both:

```suggestion
          if ( value.trim() === '' ) setThemeName( `Overlay ${timestamp()}` )
          else setThemeName( value )
```

**Review by harjot1singh — CHANGES_REQUESTED** [review]

File: src/components/editors/ExportEditor.js (hunk 2 above)
Comment (saihaj) [reply src/components/editors/ExportEditor.js]: I should get a spell checker 😩 extension

Comment (bhajneet) [reply src/components/editors/ExportEditor.js]: I use Code Spell Checker `streetsidesoftware.code-spell-checker` 😄

**Review by harjot1singh — APPROVED** [review]

**Pull request merged**